gtsfm/runner/frontend_runner.py

```diff
@@ -39,12 +39,15 @@ def run_frontend(
         v_corr_idxs_dict: verified correspondence indices for each image pair.
     """
     image_pair_indices = loader.get_valid_pairs()
-    camera_intrinsics = loader.get_all_intrinsics()
-    image_shapes = loader.get_image_shapes()
+    camera_intrinsics_graph = loader.create_computation_graph_for_intrinsics()
+    image_shapes_graph = loader.create_computation_graph_for_image_shapes()
+    
+    [image_shapes] = dask.compute(image_shapes_graph)
+    [camera_intrinsics] = dask.compute(camera_intrinsics_graph)
 
     (delayed_keypoints, delayed_putative_corr_idxs_dict,) = correspondence_generator.create_computation_graph(
         delayed_images=loader.create_computation_graph_for_images(),
-        image_shapes=loader.get_image_shapes(),
+        image_shapes=image_shapes,
```

Comment:
can the correspondence generator not take delayed shapes?

Reply:
Ideally, we should always use delayed objects that are connected with reading/storing large files. Fixed it here and created a TODO to change the function signature in a new PR.